src/pages/workspace/WorkspaceBankAccountPage.js

```diff
@@ -1,3 +1,84 @@
-const WorkspaceBankAccountPage = () => {};
+import lodashGet from 'lodash/get';
+import React from 'react';
+import {withOnyx} from 'react-native-onyx';
+import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
+import {Bank} from '../../components/Icon/Expensicons';
+import {BankArrowPink} from '../../components/Icon/Illustrations';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import Text from '../../components/Text';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import compose from '../../libs/compose';
+import BankAccount from '../../libs/models/BankAccount';
+import Navigation from '../../libs/Navigation/Navigation';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+import reimbursementAccountPropTypes from '../ReimbursementAccount/reimbursementAccountPropTypes';
+import WorkspaceSection from './WorkspaceSection';
 
-export default WorkspaceBankAccountPage;
+const propTypes = {
+    /** ACH data for the withdrawal account actively being set up */
+    reimbursementAccount: reimbursementAccountPropTypes,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    reimbursementAccount: {
+        loading: true,
+    },
+};
+
+/**
+ * We need to pop this page before we redirect or else when we goBack() we'll get a white screen
+ */
+function popAndNavigateToBankAccountRoute() {
+    Navigation.pop();
+    Navigation.navigate(ROUTES.getBankAccountRoute());
+}
+
+const WorkspaceBankAccountPage = (props) => {
+    // If we have an open bank account or no bank account at all then we will immediately redirect the user to /bank-account to display the next step
+    const state = lodashGet(props.reimbursementAccount, 'achData.state');
+    if (!lodashGet(props.reimbursementAccount, 'achData.bankAccountID') || state === BankAccount.STATE.OPEN) {
+        popAndNavigateToBankAccountRoute();
+        return null;
+    }
+
+    // Otherwise we should an interstitial page that let's them continue progress
+    return (
+        <ScreenWrapper>
+            <HeaderWithCloseButton
+                title={props.translate('workspace.common.bankAccount')}
+                onCloseButtonPress={Navigation.dismissModal}
+                onBackButtonPress={() => Navigation.goBack()}
```

Comment:
On most of these pages I am using `Navigation.navigate(ROUTES.getWorkspaceInitialRoute(policyID))` rather than `goBack()` (which works better in the cases where someone gets dropped on this page or refreshes). You'll probably need to remind me... but does this page have the policyID in the route params? If not, then `goBack()` is fine.

Reply:
Sure we can access it from here.